rir/src/compiler/util/visitor.cpp

```diff
@@ -13,8 +13,8 @@ void DominatorTreeVisitor<VisitorHelpers::IDMarker>::run(
     todo.push_back(entry);
 
     while (!todo.empty()) {
-        BB* cur = todo.front();
-        todo.pop_front();
+        BB* cur = todo.back();
```

Comment:
I think this change is also wrong. Let's take our favourite example:
```
  A
 / \
B   E
|   |
C   |
 \ /
  D
```

the `dominatorTreeNext` function will (before you changed it) give you: B,E,D for the node A.

next we want to visit B, so clearly fifo is what we want here and not lifo (like you changed it here).
But then, to have a depth first traversal we need to insert C to the front.

in short the `dominatorTreeNext` need to be inserted to the front of the todo list, but in the original order. So we need a temp list of the next nodes and then insert this list in the original order to the front of the todo list.

Reply:
or even shorter summary of the bug:
the problem is that using `dom.dominatorTreeNext(cur, [&](BB* bb) { todo.push_back(bb); });`, when you consume from the back, the immediate dominated nodes actually appear in reverse.